perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -1146,6 +1157,83 @@ public boolean isLoginReserved(PerunSession sess, String namespace, String login
 		}
 	}
 
+	@Override
+	public List<Pair<String, String>> getAllBlockedLoginsInNamespaces(PerunSession sess) {
+		return jdbc.query("select login, namespace from blocked_logins", BLOCKED_LOGINS_MAPPER);
+	}
+
+	@Override
+	public boolean isLoginBlocked(PerunSession sess, String login) {
+		Utils.notNull(login, "userLogin");
+
+		return jdbc.queryForInt("select count(1) from blocked_logins where login=?", login) == 1;
+	}
+
+	@Override
+	public boolean isLoginBlockedGlobally(PerunSession sess, String login) {
+		Utils.notNull(login, "userLogin");
+
+		return jdbc.queryForInt("select count(1) from blocked_logins where login=? and namespace is null", login) == 1;
+	}
+
+	@Override
+	public boolean isLoginBlockedForNamespace(PerunSession sess, String login, String namespace) {
+		Utils.notNull(login, "userLogin");
+
+		if (namespace == null) {
+			return this.isLoginBlockedGlobally(sess, login);
+		} else {
+			return jdbc.queryForInt("select count(1) from blocked_logins where login=? and namespace=?", login, namespace) == 1;
+		}
+	}
+
+	@Override
+	public void blockLogin(PerunSession sess, String login, String namespace) throws LoginIsAlreadyBlockedException {
+		Utils.notNull(login, "userLogin");
+
+		try {
+			int newId = Utils.getNewId(jdbc, "blocked_logins_id_seq");
+			jdbc.update("insert into blocked_logins(id, login, namespace) values (?,?,?)",
+				newId, login, namespace);
+			if (namespace == null) {
+				log.info("Login {} globally blocked", login);
+			} else {
+				log.info("Login {} blocked in namespace: {}", login, namespace);
+			}
+		} catch (DuplicateKeyException ex) {
+			if (namespace == null) {
+				throw new LoginIsAlreadyBlockedException("Login: " + login + " is already blocked globally");
+			} else {
+				throw new LoginIsAlreadyBlockedException("Login: " + login + " is already blocked in namespace: " + namespace);
+			}
+		} catch (RuntimeException ex) {
+			throw new InternalErrorException(ex);
+		}
+	}
+
+	@Override
+	public void unblockLogin(PerunSession sess, String login, String namespace) throws LoginIsNotBlockedException {
+		Utils.notNull(login, "userLogin");
+
+		try {
+			if (namespace == null) {
+				int numAffected = jdbc.update("delete from blocked_logins where login=? and namespace is null", login);
```

Comment:
I am not sure how to clearly and safely replace this branching with ternary syntax. The `is null` is not interchangeable with something like `=null` or by an empty string, am I right? And `log.info(...)` methods have a different structure, so for one of them, it doesn't make sense to put an empty string or something like that instead of the null namespace.

So how do you mean the replacement with the ternary operator? Can you give me an example, please? :face_with_head_bandage: